### PR TITLE
Analysis dat

### PR DESCRIPTION
**Pull request by vanp33**

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -2,59 +2,152 @@
...
+			}
+			return aSum -bSum;
+			});
+		for (var j= 0;j<n;j++){
+			nMost[j]= listA[j];
+		}
+		return nMost;
+	};
+	var testDateAndName = function (date,name){
+		var error = new Error();
```
Comment (jcebel) [inline index.js:33]: What for is this needed?

File: index.js [hunk 2]
```diff
@@ -2,59 +2,152 @@
...
+		error.date = date;
+		error.searchedName=name;
+		if (!date) {
+			//throw new Error('Date is empty');
+			return new SearchError('Date is empty',date, name);
+		}
+		if (date===new Date(1990,1,1)){
```
Comment (jcebel) [inline index.js:40]: This is not helpful error checking. For new Date(1991,1) I won't probably get any data. The error here needs to be different: Check if there exists content for this date in the database.

File: index.js [hunk 3]
```diff
@@ -2,59 +2,152 @@
...
+ "characterName" : "Jon Snow",
+ "date" : "2016-03-18T"
+ }
+ */
+getSentimentForName: function (json, callback) {
 
-module.exports = {
+	var date = json.date;
+	var charName = json.characterName;
+	var error = testDateAndName(date,charName);
```
Comment (jcebel) [inline index.js:92]: Please be aware that `date` is an ISO-Date-String **NOT** a Date-Object. In `testDateAndName()` you're using it as if it was a Date-Object

File: index.js [hunk 4]
```diff
@@ -2,59 +2,152 @@
...
+			return new SearchError('Date is empty',date, name);
+		}
+		if (date===new Date(1990,1,1)){
+			//error.message="For this date does no Twitterdata exist!";
+			//throw error;
+			return new SearchError('For this date does no Twitterdata exist!',date, name);
+		}
+		return;
+	};
+	var testDate = function(date){
```
Comment (jcebel) [inline index.js:47]: What for is this used?

File: index.js [hunk 5]
```diff
@@ -2,59 +2,152 @@
...
+	var endDate = json.endDate;
+	//if (testDate(startDate,charName)){console.log('Error');return;}
+	if (endDate === undefined){
+		//basiclly using getSentimentForName()
+
+	}
+	else {
+		//if(testDate(endDate,charName)){console.log('Error');return;}
+		//mongodb api here, then handle the response from mongodb
+		database.getSentimentForNameTimeframe(name,startDate,endDate,function(json){callback(json);});
```
Comment (jcebel) [inline index.js:133]: Where is here no error handling? You have to catch wrong user input and give back a meaningful errormessage. This is crucial in a project with different teams.

File: index.js (hunk 1 above)
Comment (vanp33) [inline index.js:33]: to get the elements, whose property is biggest, from the response. For example you want the 3 characters with most "posSum"(topSentiment).
For mostTalkedabout and topControversial its a little difficult, you need the sum from "posCount","negCount","nullCount" for mostTalkedAbout,
"posSum","negSum" for topControversial.
Is it clear now :)

File: index.js [hunk 6]
```diff
@@ -2,59 +2,152 @@
...
+ "startDate" : ISODate",
+ "endDate" : 'ISODate"
+ }
+ */
+getSentimentForNameTimeframe: function (json, callback) {
+	var name = json.name;
+	var startDate = json.startDate;
+	var endDate = json.endDate;
+	//if (testDate(startDate,charName)){console.log('Error');return;}
+	if (endDate === undefined){
```
Comment (jcebel) [inline index.js:126]: What for is that?

**Comment by jcebel** [comment]
Hey dat please refactor this one. 
The error-handling is not what we discussed or what I meant it to be. I started to do some comments in the first columns but later I stopped. It can be that I've misunderstood some points, but if the Travis-Tests are failing, please make absolutely sure that there are no bugs in **your** code. At the moment I wouldn't merge it. There are many things unclear and unfinished.
It does not help the others to push this unfinished.

File: index.js (hunk 1 above)
Comment (jcebel) [inline index.js:33]: No I meant the `new Error()`. You're using a different error-type later don't you?

File: index.js (hunk 2 above)
Comment (vanp33) [inline index.js:40]: do we have the api to do this ? or i must implement it?

**Pull request closed without merging**